lib/vrl/compiler/src/expression/op.rs

```diff
@@ -263,6 +290,23 @@ impl DiagnosticError for Error {
                 Label::context("this expression never resolves", rhs_span),
                 Label::context("remove this error coalesce operation", op_span),
             ],
+            MergeNonObjects { lhs_span, rhs_span } => {
+                let mut labels = Vec::new();
+                if let Some(lhs_span) = lhs_span {
+                    labels.push(Label::context(
+                        "this expression must resolve to an object",
+                        lhs_span,
+                    ));
```

Comment:
I propose changing this to `Label::primary`.

Basically, every error should (almost) always have at least _one_ primary span, which is coloured red, and points to the issue.

Then, if need be, one or more context spans can be used to explain what's going on and how the user can solve the issue.